### PR TITLE
Use appLaunchedInForeground to determine invalid app start data on Android

### PR DESCRIPTION
**Pull request by antonis**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description

Checks `appLaunchedInForeground` to determine invalid app start data on Android.
Returns `null` in the `fetchNativeAppStart` promise when the app has not started in foreground. The cases that a `null` is returned are already handled in the code due to [the iOS implementation that returns `nil` for invalid measurements](https://github.com/getsentry/sentry-react-native/blob/main/ios/RNSentry.mm#L394). 

Note: [There is already a check in place](https://github.com/getsentry/sentry-react-native/blob/main/src/js/tracing/reactnativetracing.ts#L497) for launches longer [than 1 min](https://github.com/getsentry/sentry-react-native/blob/main/src/js/tracing/reactnativetracing.ts#L141).

## :bulb: Motivation and Context

Fixes https://github.com/getsentry/sentry-react-native/issues/3995

## :green_heart: How did you test it?
Added tests for the [`RNSentryModuleImpl.fetchNativeAppStart`](https://github.com/getsentry/sentry-react-native/pull/4146/files#diff-eb98fb9e4fc820a10dba9ff587d350f39ea67f425391fd46f872992bc54c84f6) function logic and relied on existing unit test:
- [Does not instrument app start if fetchNativeAppStart returns null](https://github.com/getsentry/sentry-react-native/blob/main/test/tracing/reactnativetracing.test.ts#L816)

and the [`sentry-java` tests](https://github.com/getsentry/sentry-java/pull/3550).

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Comment by krystofwoldrich** [comment]
Thank you, I'll review this in the afternoon.

File: RNSentryAndroidTester/app/src/test/java/io/sentry/react/RNSentryModuleImplTest.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    private lateinit var module: RNSentryModuleImpl
+    private lateinit var promise: Promise
+    private lateinit var logger: ILogger
+    private var argumentsMock: MockedStatic<Arguments>? = null
+
+    @Captor
+    private lateinit var writableMapCaptor: ArgumentCaptor<WritableMap>
+
+
```
Comment (lucas-zimerman) [inline RNSentryAndroidTester/app/src/test/java/io/sentry/react/RNSentryModuleImplTest.kt:36]: NIT
```suggestion
```

**Review by lucas-zimerman — APPROVED** [review]
looks good to me, thank you for adding the Native tests!

**Comment by krystofwoldrich** [comment]
Thank you @lucas-zimerman for the review and thanks @antonis for the impl and the tests. I bit late but looks good!

**Pull request merged**